pkg/agent/config/node_config.go

```diff
@@ -250,11 +254,35 @@ func (nc *NetworkConfig) NeedsTunnelToPeer(peerIP net.IP, localIP *net.IPNet) bo
 	return nc.TrafficEncapMode == TrafficEncapModeEncap || (nc.TrafficEncapMode == TrafficEncapModeHybrid && !localIP.Contains(peerIP))
 }
 
+func (nc *NetworkConfig) NeedsTunnelInterface() bool {
+	return nc.TrafficEncapMode.SupportsEncap() || nc.EnableMulticlusterGW
+}
+
 // NeedsDirectRoutingToPeer returns true if Pod traffic to peer Node needs a direct route installed to the routing table.
 func (nc *NetworkConfig) NeedsDirectRoutingToPeer(peerIP net.IP, localIP *net.IPNet) bool {
 	return (nc.TrafficEncapMode == TrafficEncapModeNoEncap || nc.TrafficEncapMode == TrafficEncapModeHybrid) && localIP.Contains(peerIP)
 }
 
+func (nc *NetworkConfig) CalculateMTUDeduction(isIPv6 bool) int {
+	var mtuDeduction int
+	// When Multi-cluster Gateway is enabled, we need to reduce MTU for potential cross-cluster traffic.
+	if nc.TrafficEncapMode.SupportsEncap() || nc.EnableMulticlusterGW {
+		if nc.TunnelType == ovsconfig.VXLANTunnel {
+			mtuDeduction = VXLANOverhead
```

Comment:
Could you check if VXLANOverhead, IPv6ExtraOverhead, etc. can be private now?

Reply:
Yeah, they can be private, renamed them.